serve/plugin.go

```diff
@@ -142,11 +145,15 @@ func (s *PluginServe) newCmdPluginServe() *cobra.Command {
 
 			if otelEndpoint != "" {
 				resource := newResource(s.plugin)
-				opts := []otlptracehttp.Option{
+				traceOptions := []otlptracehttp.Option{
```

Comment:
I think `if otelEndpoint != "" {` should be extracted into a separate`setupOTel` func